Sources/NIO/Bootstrap.swift

```diff
@@ -354,11 +356,11 @@ private extension Channel {
 public final class ClientBootstrap {
 
     private let group: EventLoopGroup
-    private var channelInitializer: ((Channel) -> EventLoopFuture<Void>)?
+    private var channelInitializer: Optional<(Channel) -> EventLoopFuture<Void>>
```

Comment:
Nit, maybe out of scope, now that this line is a little more "complex" a typealias might be nice

Reply:
@Davidde94 hmm, what name would you propose for a typealias for `(Channel) -> EventLoopFuture<Void>`?

Reply:
`ChannelInitializer`?

Reply:
doesn't that make it harder to see what that is? CC @Lukasa I'm open to making it `Optional<ChannelInitializer>` but I think that's more opaque than `Optional<(Channel) -> ELF<Void>>` but maybe it's me?

Reply:
I don't love `ChannelInitializer` as a typealias: `ChannelInitializerCallback` might be better.

Reply:
Either sound good to me - note this was a readability nit, I think the good is good to go regardless :)

Reply:
@Davidde94 / @Lukasa fixed.